### PR TITLE
docs/user-guide/pod-security-policy: list all available PSP fields

### LINKED ISSUE #2116: Issue with /docs/user-guide/pod-security-policy/
**This is a...** 

- [x] Feature Request
- [ ] Bug Report

**Problem:**
`PodSecurityPolicy` has many fields and I see that most of them not listed in the doc. As a user I want to be able to create my custom `PodSecurityPolicy` after reading this doc. Even if user decided to read the initial proposal it will found that it also a bit outdated.

**Page to Update:**
http://kubernetes.io/docs/user-guide/pod-security-policy/

### PR DESCRIPTION
**Pull request by php-coder**
Fixes #2116

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/3012)

### CODE REVIEW

**Comment by php-coder** [comment]
@pweil- Is there a way to preview this changes? Maybe generate the docs locally (I didn't search yet.)

**Comment by pweil-** [comment]
> @pweil- Is there a way to preview this changes? Maybe generate the docs locally (I didn't search yet.)

you can stage locally with docker: https://kubernetes.io/docs/contribute/stage-documentation-changes/

File: docs/user-guide/pod-security-policy/index.md [hunk 1]
```diff
@@ -21,15 +21,22 @@ actions that a pod can perform and what it has the ability to access. The
...
+| Controlling the usage of volume types                         | `volumes`                  |
+| The use of host networking                                    | `hostNetwork`              |
+| The use of host ports                                         | `hostPorts`                |
+| The use of host's PID namespace                               | `hostPID`                  |
+| The use of host's IPC namespace                               | `hostIPC`                  |
+| The SELinux context of the container                          | `seLinux`                  |
+| The user ID                                                   | `runAsUser`                |
+| Configuring allowable supplemental groups                     | `supplementalGroups`       |
+| Allocating an FSGroup that owns the pod's volumes             | `fSGroup`                  |
+| Requiring the use of a read only root file system             | `readOnlyRootFilesystem`   |
```
Comment (pweil-) [inline docs/user-guide/pod-security-policy/index.md]: Should also include info about allowed annotations for alpha features:

```
// The annotation key specifying the default seccomp profile.
	DefaultProfileAnnotationKey = "seccomp.security.alpha.kubernetes.io/defaultProfileName"
	// The annotation key specifying the allowed seccomp profiles.
	AllowedProfilesAnnotationKey = "seccomp.security.alpha.kubernetes.io/allowedProfileNames"
```

```
// SysctlsPodSecurityPolicyAnnotationKey represents the key of a whitelist of
	// allowed safe and unsafe sysctls in a pod spec. It's a comma-separated list of plain sysctl
	// names or sysctl patterns (which end in *). The string "*" matches all sysctls.
	SysctlsPodSecurityPolicyAnnotationKey string = "security.alpha.kubernetes.io/sysctls"
```

```
// The annotation key specifying the default AppArmor profile.
	DefaultProfileAnnotationKey = "apparmor.security.beta.kubernetes.io/defaultProfileName"
	// The annotation key specifying the allowed AppArmor profiles.
	AllowedProfilesAnnotationKey = "apparmor.security.beta.kubernetes.io/allowedProfileNames"
```

Comment (pweil-) [inline docs/user-guide/pod-security-policy/index.md]: (search annotation in the `security/podsecuritypolicy` directory to find which ones are used in strategies or providers)

**Comment by php-coder** [comment]
PTAL @pweil-

**Comment by chenopis** [comment]
@pweil- Ping.

**Comment by pweil-** [comment]
This is waiting on update with reference to the comment on https://github.com/kubernetes/kubernetes.github.io/issues/3174 unless someone disagrees or if we have an official policy wrt the statement about annotations (which overrides my previous request for change) :smile:

**Comment by php-coder** [comment]
The code is here so I don't understand why we're not documenting it. How user will know about possible features then?

**Comment by pweil-** [comment]
> The code is here so I don't understand why we're not documenting it. How user will know about possible features then?

I'm not disagreeing.  Best thing to do is take the conversation to the other issue and work out the reasoning there.  Leave this as-is until then.

**Comment by php-coder** [comment]
@pweil- Would you approve this PR if I remove mention about annotations? Documentation is updating very frequently I don't want to keep this PR open for a long time.

**Comment by pweil-** [comment]
If you'd like to leave it out for now while the discussion is ongoing then this is LGTM

**Comment by php-coder** [comment]
I've adapted my changes to the latest master (where this file were moved to other place).

@pweil- PTAL.

**Comment by php-coder** [comment]
@chenopis It's ready to merge.

**Pull request merged**